src/profiles_management/pmr/classes.py

```diff
@@ -0,0 +1,133 @@
+Pythonic way, rather than a dict of vavlues.
+
+The classes should also provide some common helper methods for
+traversing the information of the PMR, but not functions for
+directly affecting the K8s cluster based on the PMR.
+
+This is similar to how the K8s python clients have classes
+for representing the objects and different methods for doing operations
+with them.
+"""
+
+import logging
+from dataclasses import dataclass
+from enum import Enum
+from typing import Any, Dict, List
+
+import jsonschema
+
+from profiles_management.pmr.schema import RESOURCE_QUOTA_SCHEMA
+
+log = logging.getLogger(__name__)
+
+
+class UserKind(Enum):
+    """Class representing the kind of the user."""
+
+    USER = "user"
+    SERVICE_ACCOUNT = "service-account"
+
+
+class ContributorRole(Enum):
+    """Class representing the role of the user."""
+
+    ADMIN = "admin"
+    EDIT = "edit"
+    VIEW = "view"
+
+
+@dataclass
+class Contributor:
+    """Class representing which users should have access to a Profile."""
+
+    name: str
+    role: ContributorRole
+
+
+@dataclass
+class Owner:
+    """Class representing the owner field of a Profile."""
+
+    name: str
+    kind: UserKind
+
+
+@dataclass
+class Profile:
+    """Class representing a Profile and its Contributors."""
+
+    name: str
+    owner: Owner
+    resources: Dict[str, Any]
+    contributors: List[Contributor]
+
+    # https://docs.python.org/3/library/dataclasses.html#post-init-processing
+    def __post_init__(self):
+        """Validate resourceQuota after values have been initialised."""
+        log.info("Validating ResourceQuota for Profile: %s", self.name)
+        jsonschema.validate(self.resources, RESOURCE_QUOTA_SCHEMA)
+        log.info("ResourceQuota is valid.")
+
+    def has_contributor(self, name: str, role: ContributorRole) -> bool:
+        """Check if the Profile has a contributor with specific role."""
+        for contributor in self.contributors:
+            if contributor.name != name:
+                continue
+
+            if contributor.role != role:
+                continue
+
+            log.info("Profile %s has contributor (%s, %s)", self.name, name, role.value)
+            return True
+
+        log.info("Profile %s doesn't have contributor (%s, %s)", self.name, name, role.value)
+        return False
+
+
+class ProfilesManagementRepresentation:
+    """A class representing the Profiles and Contributors."""
+
+    profiles: dict[str, Profile] = {}
+
+    def __init__(self, profiles: List[Profile] = []):
+        """Initialise based on a list of Profiles.
+
+        If a list of Profiles is given, then the internal dict will be initialised
+        based on this list.
+        """
+        for profile in profiles:
+            self.add_profile(profile)
+
+    def has_profile(self, name: str | None) -> bool:
+        """Check if given Profile name is part of the PMR."""
+        return name in self.profiles
+
+    def add_profile(self, profile: Profile):
```

Comment:
Should `add_profile` receive also a list of profiles? This way we'll be able to pass a list of `Profile` objects that can just be added to the PMR instead of adding them one by one.

Reply:
I'd say no, since this is covered by the `__ini__` function

Reply:
If we are okay with passing one by one, then that's alright.